src/psyclone/domain/gocean/nodes/gocean_extract_node.py

```diff
@@ -0,0 +1,257 @@
+                call = CallGen(prog,
+                               "psy_data%ReadVariable(\"{0}\", {0})"
+                               .format(var_name))
+                prog.add(call)
+            elif is_input:
+                # Now must be input and output:
+                # First read the pre-variable
+                call = CallGen(prog,
+                               "psy_data%ReadVariable(\"{0}\", {0})"
+                               .format(var_name))
+                prog.add(call)
+                # Then declare the post variable, and and read its values
+                # (ReadVariable will also allocate it)
+                decl = DeclGen(prog, "real", [var_name+post_suffix],
+                               dimension=":,:", kind="8", allocatable=True)
+                prog.add(decl)
+                call = CallGen(prog,
+                               "psy_data%ReadVariable(\"{0}\", {0})"
+                               .format(var_name+post_suffix))
+                prog.add(call)
+            else:
+                # Now the variable is output only. We need to read the
+                # post variable in, and allocate the pre variable with
+                # the same size as the post
+                decl = DeclGen(prog, "real", [var_name+post_suffix],
+                               dimension=":,:", kind="8", allocatable=True)
+                prog.add(decl)
+                call = CallGen(prog,
+                               "psy_data%ReadVariable(\"{0}\", {0})"
+                               .format(var_name+post_suffix))
+                prog.add(call)
+                decl = DeclGen(prog, "real", [var_name], kind="8",
+                               dimension=":,:", allocatable=True)
+                prog.add(decl)
+                alloc = AllocateGen(prog, [var_name,
+                                           "mold={0}".format(var_name +
+                                                             post_suffix)])
+                prog.add(alloc)
+                # Initialise the variable with 0, since it might contain
+                # values that are not set at all (halo regions, or a
+                # kernel might not set all values). This way the array
+                # comparison with the post value works as expected
+                assign = AssignGen(prog, var_name, "0.0")
+                prog.add(assign)
+
+        # Now add the region that was extracted here:
+        prog.add(CommentGen(prog, ""))
+        prog.add(CommentGen(prog, " RegionStart"))
+
+        # For the driver we have to re-create the code, but
+        # the arguments are not fields anymore, but simple arrays.
+        # So we need to make sure that the field parameters
+        # are changed from "fld%data" to just "fld", and similar
+        # for all grid properties. This is
+        # achieved by temporary changing the value of the
+        # all grid properties (e.g. "{0}%data") to just "{0}".
```

Comment:
No, this is correct: if the field name is ``a_field``, then the first format statement will create ``a_field%data``. After changing the format, it will just create ``a_field``, which is the name of the array in the stand-alone Fortran driver.